fastembed/splitter.py

```diff
@@ -0,0 +1,206 @@
+# Custom implementation based on the Langchain text splitter
+# Reference: https://python.langchain.com/docs/modules/data_connection/document_transformers/recursive_text_splitter
+
+import logging
+import re
+from abc import ABC, abstractmethod
+from typing import (
+    Callable,
+    Iterable,
+    List,
+    Optional,
+)
+from tokenizers import Tokenizer
+
+
+logger = logging.getLogger(__name__)
+
+
+def _split_text_with_regex(text: str, separator: str, keep_separator: bool) -> List[str]:
+    # Now that we have the separator, split the text
+    if separator:
+        if keep_separator:
+            # The parentheses in the pattern keep the delimiters in the result.
+            _splits = re.split(f"({separator})", text)
+            splits = [_splits[i] + _splits[i + 1] for i in range(1, len(_splits), 2)]
+            if len(_splits) % 2 == 0:
+                splits += _splits[-1:]
+            splits = [_splits[0]] + splits
+        else:
+            splits = re.split(separator, text)
+    else:
+        splits = list(text)
+    return [s for s in splits if s != ""]
+
+
+class TextSplitter(ABC):
+    """Interface for splitting text into chunks."""
+
+    def __init__(
+        self,
+        chunk_size: int = 4000,
+        chunk_overlap: int = 200,
+        length_function: Callable[[str], int] = len,
+        keep_separator: bool = False,
+        strip_whitespace: bool = True,
+    ) -> None:
+        """Create a new TextSplitter.
+
+        Args:
+            chunk_size: Maximum size of chunks to return
+            chunk_overlap: Overlap in characters between chunks
+            length_function: Function that measures the length of given chunks
+            keep_separator: Whether to keep the separator in the chunks
+            strip_whitespace: If `True`, strips whitespace from the start and end of
```

Comment:
We mention and measure size and overlap in character or tokens? We should maintain consistency across the codebase please?